app/services/npq/create_participant_outcome.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module NPQ
+  class CreateParticipantOutcome
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :cpd_lead_provider
+    attribute :participant_external_id
+    attribute :course_identifier
+    attribute :state
+    attribute :completion_date, :date
+
+    validates :cpd_lead_provider, :participant_external_id, presence: true
+    validates :completion_date, presence: { message: I18n.t(:missing_completion_date) }
```

Comment:
ah just missed this: could we also validate date format as well?

Reply:
Hi @cwrw it's been addressed in https://github.com/DFE-Digital/early-careers-framework/pull/2809/commits/70b1b19fb85e307fb17ec96fb9d95b6d936c51d6 tks for your review.